### PR TITLE
Fix mpich build on macOS due to dependencies on libpciaccess.

### PR DESCRIPTION
**Pull request by adamqc**

### CODE REVIEW

File: var/spack/repos/builtin/packages/mpich/package.py [hunk 1]
```diff
@@ -76,7 +79,7 @@ class Mpich(AutotoolsPackage):
 
     depends_on('libfabric', when='netmod=ofi')
 
-    depends_on('libpciaccess')
+    depends_on('libpciaccess', when="+pci")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/mpich/package.py]: @adamqc If you think MacOS users would never need `libpciaccess` for `mpich`, you can condition the when clause on the platform too:

`depends_on('libpciaccess', when="platform=darwin")`

In which case the variant is not needed

Comment (adamqc) [reply var/spack/repos/builtin/packages/mpich/package.py]: Sorry for the delayed reply. I think it should be
 ```depends_on('libpciaccess', when="platform!=darwin")```
But I get `==> Error: Invalid character`, it seems that we can not use `!=` in `when` argument. Do you have any suggestions?

Comment (scheibelp) [reply var/spack/repos/builtin/packages/mpich/package.py]: > I think it should be
> `depends_on('libpciaccess', when="platform!=darwin")`
> But I get `==> Error: Invalid character`, it seems that we can not use `!=` in `when` argument.

You are correct: currently "not" is not supported in Spack spec syntax constraints, and I got the constraint wrong.

So for now I think you have the best solution

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**